pallets/staking/src/lib.rs

```diff
@@ -2345,24 +2482,29 @@ impl<T: Trait> Module<T> {
 
     /// Actually make a payment to a staker. This uses the currency's reward function
     /// to pay the right payee for the given staker account.
-    fn make_payout(stash: &T::AccountId, amount: BalanceOf<T>) -> Option<PositiveImbalanceOf<T>> {
+    fn make_payout(stash: &T::AccountId, amount: Balance) -> Option<PositiveImbalanceOf<T>> {
         let dest = Self::payee(stash);
+        let stash_liquidity_token = match Self::get_stash_liquidity_token(stash) {
+            Some(liquidity_token) => liquidity_token,
+            None => return None,
+        };
         match dest {
-            RewardDestination::Controller => Self::bonded(stash)
-                .and_then(|controller| Some(T::Currency::deposit_creating(&controller, amount))),
-            RewardDestination::Stash => T::Currency::deposit_into_existing(stash, amount).ok(),
-            RewardDestination::Staked => Self::bonded(stash)
-                .and_then(|c| Self::ledger(&c).map(|l| (c, l)))
-                .and_then(|(controller, mut l)| {
-                    l.active += amount;
-                    l.total += amount;
-                    let r = T::Currency::deposit_into_existing(stash, amount).ok();
-                    Self::update_ledger(&controller, &l);
-                    r
-                }),
-            RewardDestination::Account(dest_account) => {
-                Some(T::Currency::deposit_creating(&dest_account, amount))
+            RewardDestination::Controller => Self::bonded(stash).and_then(|controller| {
```

Comment:
the rewards should be payed in Mantaga tokens, not in liquidity tokens

Reply:
Right! Thanks for catching that. will change.

Reply:
Changed.